### PR TITLE
Ft api endpoint syntax fixes #161400939

### PR DESCRIPTION
**Pull request by ibraheemkabir**
### What does this PR do?
- M,oves the array operations into the main controller
- Removes unnecessary model for array types
- Moves "test" environment declaration from test file to test script
- Changes some variable names to meaningful ones.
- Changes array find operation to findbyindex operation for optimal perfomance
- Changes unnecessary async functions to regular class methods.
- Changes 'allproducts' endpoint to 'getproducts'
- Updates test coverage script on develop branch

### Description of Task to be completed?
- The array operations are moved into the main controller to get rid of unnecessary external querying.
- The array model which assigns a type to fields in the array object is not appropriate thus it is removed.
- Declaration of test environment in test file leads to multiple environment creation thus this is to be moved to the test script.
- The find method used in the array is inappriopriate as it leads to errors thus it is changed to findbyindex method

### How should this be manually tested?
Test the endpoints with postman

### My background context you want to provide?
All changes are made as requested by LFA for optimal api perfomance.

### What are the relevant pivotal tracker stories?(if applicable)
#161400939

### CODE REVIEW

File: server/controllers/store.js [hunk 1]
```diff
@@ -1,25 +1,23 @@
-import express from 'express';
-
+/* eslint-disable class-methods-use-this */
```
Comment (amarachukwu-agbo) [inline server/controllers/store.js]: Please remove the eslint-disable here. Look for a solution for this eslint error

File: server/controllers/store.js [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
+    const {name, category, price, size } = req.body;
+    const newProduct = product.push(new Prods(product.length+1, name, category, price, size));
       success: 'true',
       message: 'product added successfuly',
       product,
     });
   }
 
+
   static async allProducts(req, res) {
```
Comment (amarachukwu-agbo) [inline server/controllers/store.js]: I believe you should name this getAllProducts

File: server/controllers/store.js [hunk 3]
```diff
@@ -1,25 +1,23 @@
...
-import queries from '../models/queries';
-
-export default class Products {
 
-  static async addProduct(req, res) {
-    const newProduct = queries.addProduct;
-    const products = await (newProduct);
-    return res.status(201).json({
+class Products {
+  addProduct(req, res) {
```
Comment (amarachukwu-agbo) [inline server/controllers/store.js]: Why are you using static keyword in some method and not others?

File: server/controllers/store.js [hunk 4]
```diff
@@ -42,19 +41,18 @@ export default class Products {
...
       success: 'false',
       message: 'product not found',
-      arr,
     });
   }
 
-  static async getproduct(req, res) {
+  getproduct(req, res) {
     const id = parseInt(req.params.id, 10);
     const arr = product.find(c => c.id === parseInt(req.params.id, 10));
```
Comment (amarachukwu-agbo) [inline server/controllers/store.js]: Please change arr to a more descriptive name

File: server/controllers/store.js [hunk 5]
```diff
@@ -63,32 +61,21 @@ export default class Products {
...
   }
 
-
-  static async updateproduct(req, res) {
-    const update = queries.updateproduct;
-    const arr = await (update);
-    if (!arr) {
-      return res.status(404).json({
+  updateproduct(req, res) {
+    const arr = product.findIndex(c => c.id === parseInt(req.params.id, 10));
```
Comment (amarachukwu-agbo) [inline server/controllers/store.js]: Here too, change variable to be more descriptive

File: server/controllers/store.js [hunk 6]
```diff
@@ -100,43 +87,47 @@ export default class Products {
...
   }
 
-  static async addcategory(req, res) {
-    const { Category } = req.body ;
-    const newcategory = catigory.push(new category(catigory.length,Category));
-    const categorys = await (newcategory);
-    return res.status(201).json({
+  addcategory(req, res) {
+    const { Category } = req.body;
+    const newcategory = category.push(new categoryarray(category.length+1, Category));
```
Comment (amarachukwu-agbo) [inline server/controllers/store.js]: Please use camelCase convention for naming variables

File: server/controllers/store.js [hunk 7]
```diff
@@ -100,43 +87,47 @@ export default class Products {
...
       message: 'category added successfuly',
-      catigory,
+      category,
     });
   }
 
-  static async getcategories(req, res) {
-    return res.status(200).json({
+
+  getcategories(req, res) {
```
Comment (amarachukwu-agbo) [inline server/controllers/store.js]: Use camelCase for naming this method

File: server/controllers/store.js [hunk 8]
```diff
@@ -100,43 +87,47 @@ export default class Products {
...
       success: 'true',
       message: 'orders retrieved successfuly',
       Order,
     });
   }
+
+
 }
+
+const Queries = new Products();
```
Comment (amarachukwu-agbo) [inline server/controllers/store.js]: Why are you exporting this as default?

File: server/middleware/productauth.js [hunk 9]
```diff
@@ -2,8 +2,8 @@ import express from 'express';
 
 import products from '../models/productsModel';
 
-export default class Productauth{
-    static async addproduct(req, res, next) {
+class Productauth{
```
Comment (amarachukwu-agbo) [inline server/middleware/productauth.js:5]: Why did you remove static keyword from the methods?

File: server/middleware/productauth.js [hunk 10]
```diff
@@ -54,23 +54,32 @@ export default class Productauth{
         return next();
     }
 
-    static async updateproduct(req, res, next) {
-        const { name} = req.body;
-        const { size } = req.body;
-        if (!name) {
+     updateproduct(req, res, next) {
```
Comment (amarachukwu-agbo) [inline server/middleware/productauth.js]: Did you have to repeat this code? Since it already existed, I thought you would reuse the already existing code

File: server/test/test.js [hunk 11]
```diff
@@ -78,6 +81,9 @@ describe('/POST products', () => {
         done();
       });
   });
+});
+
+describe('/GET products', () => {
 
   it('it should get a particular product', (done) => {
```
Comment (amarachukwu-agbo) [inline server/test/test.js:132]: You did not test for 404 when the product was not found as we discussed yesterday even for the other endpoints

**Review by amarachukwu-agbo — CHANGES_REQUESTED** [review]

File: server/middleware/productauth.js (hunk 9 above)
Comment (ibraheemkabir) [reply server/middleware/productauth.js:5]: removed it because i was exporting the class object  but seems unnecesary so i've reverted it back

File: server/test/test.js (hunk 11 above)
Comment (ibraheemkabir) [reply server/test/test.js:132]: working on this will commit shortly

File: server/controllers/store.js (hunk 1 above)
Comment (ibraheemkabir) [reply server/controllers/store.js]: error solved and resolved.

File: server/controllers/store.js (hunk 2 above)
Comment (ibraheemkabir) [reply server/controllers/store.js]: renamed

File: server/routes/products.js [hunk 12]
```diff
@@ -8,15 +8,15 @@ import queries from '../models/queries';
...
+router.get('/products/:id', auth.getProduct , products.getProduct);
+router.get('/order',auth.allOrders, products.getAllOrders);
 router.post('/order',auth.newOrder, products.newOrder);
-router.get('/category', products.getcategories);
-router.post('/category', products.addcategory);
+router.get('/category', products.getAllCategories);
+router.post('/category',auth.newCategory, products.addCategory);
 
 export default router;
 module.exports = router;
```
Comment (amarachukwu-agbo) [inline server/routes/products.js]: Please remove the module.exports since you are already using export default

**Comment by amarachukwu-agbo** [comment]
@ibraheemkabir , for your PR description, nice attempt! However, fix the mark down issue for the different headings. Also begin each sentence with a capital letter. When you are done implementing the feedback, merge and notify me.
